src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -815,13 +816,20 @@ private static Task LogDowngradeWarningsAsync(IEnumerable<RestoreTargetGraph> gr
 
                             var message = string.Format(
                                     CultureInfo.CurrentCulture,
-                                    Strings.Log_DowngradeWarning,
+                                    isCpvmEnabled ? Strings.Log_CPVM_DowngradeError : Strings.Log_DowngradeWarning,
                                     downgraded.Key.Name,
                                     fromVersion,
                                     toVersion)
                                 + $" {Environment.NewLine} {downgraded.GetPathWithLastRange()} {Environment.NewLine} {downgradedBy.GetPathWithLastRange()}";
 
-                            messages.Add(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1605, message, downgraded.Key.Name, graph.TargetGraphName));
+                            if (isCpvmEnabled)
+                            {
+                                messages.Add(RestoreLogMessage.CreateError(NuGetLogCode.NU1109, message, downgraded.Key.Name, graph.TargetGraphName));
```

Comment:
Currently all the downgrades for CPVM are errors. My thought was to not differentiate between the types of dependencies. Why would be a differentiation?

Reply:
Because "normal" downgrade errors are resolved by adding a direct reference to the package. CPVM downgrades are resolved possibly by adding a PackageVersion to the package, if one doesn't already exist, or upgrading the version if a PackageVersion does already exist for the package. They have potentially different causes and resolutions.

Is transient version pinning always-on, or is it another opt-in feature? If it's opt-in, that's another reason to detect it differently.

Reply:
Just adding here that I agree with Andy.
The resolution method is different, so the messages should be slightly different as well. 
In 1605 we say reference the package directly,
> Reference the package directly from the project to select a different version.

In CPVM we should say that the package version should be lifted.

Reply:
Updated the code to differentiate.

Reply:
Updated the code to differentiate.